packages/ra-ui-materialui/src/field/ReferenceField.spec.tsx

```diff
@@ -612,6 +615,23 @@ describe('<ReferenceField />', () => {
             });
         });
     });
+    describe('Security', () => {
+        it('should render a link only when users have access to the requested action for the referenced resource', async () => {
+            render(<AccessControl />);
+            await waitFor(
+                () => {
+                    expect(
+                        screen
+                            .getByText('Lewis Carroll', {
+                                selector: 'a > span',
+                            })
+                            .parentElement?.getAttribute('href')
+                    ).toEqual('/authors/5/show');
```

Comment:
Instead of testing the link value, click on it and check that you're on the right page.